File: cc/Singleton/Base/inputclient.cc
#include "input.h"
#include "inputfactory.h"

int main(int argc, char* argv[])
{
    Input* input = InputFactory::instance()->create(FILE_INPUT);
    input->read();
    delete input;

    input = InputFactory::instance()->create(CONSOLE_INPUT);
    input->read();
    delete input;

    input = InputFactory::instance()->create(NET_INPUT);
    input->read();
    delete input;

    return 0;
}

File: python/Singleton/Base2/inputclient.py
"""
inputclient.py
"""

from input import *
from inputfactory import *

def main():
    input = InputFactory.instance().create("file")
    input.read()

    input = InputFactory.instance().create("console")
    input.read()

    input = InputFactory.instance().create("net")
    input.read()

if __name__ == '__main__':
    main()

